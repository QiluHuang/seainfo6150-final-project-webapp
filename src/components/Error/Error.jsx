import React from 'react'
import { Switch, Route, Redirect, NavLink, Link, useRouteMatch, useHistory } from "react-router-dom";
import styles from './Error.module.css';

import Button from '../Buttons/Button/Button.jsx';
import Home from '../../views/Home/Home.jsx';


const Error = () => {
    let history = useHistory();
    const text = "GO TO HOMEPAGE";

    function onClick() {
        history.push("/");
    }

    return (
        <div className={styles.notfound}>
            <div className={styles.notfound_404}>
                <img src="https://www.clintar.com/images/error-404.png"></img>
            </div>
            <div className={styles.pageNotfound}>PAGE NOT FOUND</div>
            <div className={styles.explanation}>The page you are looking for might have been removed or is temporaily unavailable</div>
            <div className={styles.button}><Button buttonText={text} onClick={onClick}/></div>
        </div>
    )
}

export default Error
