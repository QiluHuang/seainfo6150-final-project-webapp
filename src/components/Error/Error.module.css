.notfound {
    margin: 200px 0 30px;
    display: flex;
    flex-direction: column;
    align-items: center;
}



.pageNotfound {
    margin: 30px 0 30px;
    font-size: 2em;
    font-family: Arial;
}

.explanation {
    font-family: Arial, Helvetica, sans-serif;
}

.button {
    margin: 30px 0 30px;
}