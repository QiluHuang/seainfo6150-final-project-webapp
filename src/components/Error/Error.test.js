import React from "react";
import { render } from "@testing-library/react";
import Error from "./Error.jsx";

describe("Error tests", () => {
    it("renders the Error component correctly", () => {
    const { container } = render(<Error />);
    expect(container).toMatchSnapshot();
    });
});