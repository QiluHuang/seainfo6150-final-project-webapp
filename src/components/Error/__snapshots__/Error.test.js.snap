// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error tests renders the Error component correctly 1`] = `
<div>
  <div
    class="notfound"
  >
    <div
      class="notfound_404"
    >
      <img
        src="https://www.clintar.com/images/error-404.png"
      />
    </div>
    <div
      class="pageNotfound"
    >
      PAGE NOT FOUND
    </div>
    <div
      class="explanation"
    >
      The page you are looking for might have been removed or is temporaily unavailable
    </div>
    <div
      class="button"
    >
      <button
        class="button"
      >
        GO TO HOMEPAGE
      </button>
    </div>
  </div>
</div>
`;
